pkg/generator/generator.go

```diff
@@ -132,44 +132,58 @@ func Generate(node *v1alpha1.Node) error {
 
 				node.Value = caCertPEM
 			case v1alpha1.TypeKeyPair:
-				// get private key's PEM value
-				rootCAPrivateKeyPEM, err := getValueFromParent(node.AliasConfig.SignedWithPath, node)
-				if err != nil {
-					return err
-				}
+				var certAndKeyPEM, certPEM, keyPEM []byte
+				var err error
+				if !node.AliasConfig.SharedCert {
+					// get private key's PEM value
+					rootCAPrivateKeyPEM, err := getValueFromParent(node.AliasConfig.SignedWithPath, node)
+					if err != nil {
+						return err
+					}
 
-				// get private key's CA PEM value
-				rootCAPEM := []byte{}
-				for _, parentNode := range node.Parents {
-					if memorystore.Equal(node.AliasConfig.SignedWithPath, parentNode.Path) {
-						rootCAPEM, err = getValueFromParent(parentNode.KeyConfig.CAPath, parentNode)
-						if err != nil {
-							return err
+					// get private key's CA PEM value
+					rootCAPEM := []byte{}
+					for _, parentNode := range node.Parents {
+						if memorystore.Equal(node.AliasConfig.SignedWithPath, parentNode.Path) {
+							rootCAPEM, err = getValueFromParent(parentNode.KeyConfig.CAPath, parentNode)
+							if err != nil {
+								return err
+							}
+							break
 						}
-						break
 					}
-				}
-				if len(rootCAPEM) == 0 {
-					return errors.New("Failed to find root CA PEM value")
-				}
+					if len(rootCAPEM) == 0 {
+						return errors.New("Failed to find root CA PEM value")
+					}
 
-				// generate
-				certAndKeyPEM, certPEM, keyPEM, err := GenerateSignedCertPEM(
-					rootCAPEM,
-					rootCAPrivateKeyPEM,
-					node.AliasConfig.Algorithm,
-					node.AliasConfig.CommonName,
-					node.AliasConfig.Sans,
-				)
-				if err != nil {
-					return err
+					// generate
+					certAndKeyPEM, certPEM, keyPEM, err = GenerateSignedCertPEM(
+						rootCAPEM,
+						rootCAPrivateKeyPEM,
+						node.AliasConfig.Algorithm,
+						node.AliasConfig.CommonName,
+						node.AliasConfig.Sans,
+					)
+					if err != nil {
+						return err
+					}
+				} else if node.AliasConfig.SharedCert {
+					certAndKeyPEM, certPEM, keyPEM, err = GenerateSharedCertPEM(node.AliasConfig.CommonName)
+					if err != nil {
+						return errors.WithStack(err)
+					}
+				} else {
+					return errors.New("Cert should be either SharedCert or Signed")
 				}
 
 				// add to keystore
 				storePassword, err := getValueFromParent(node.KeyConfig.StorePassPath, node)
 				if err != nil {
 					return err
 				}
+				fmt.Printf("----------%s----------\n", node.AliasConfig.CommonName)
```

Comment:
debug logs? need to use logging libs instead

Reply:
no that's something I meant to remove.